packages/flutter/lib/src/cupertino/context_menu.dart

```diff
@@ -0,0 +1,1224 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math' as math;
+import 'dart:ui' as ui;
+import 'package:flutter/gestures.dart' show kMinFlingVelocity, kLongPressTimeout;
+import 'package:flutter/rendering.dart';
+import 'package:flutter/scheduler.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter/widgets.dart';
+import 'context_menu_action.dart';
+
+// The scale of the child at the time that the ContextMenu opens.
+// This value was eyeballed from a physical device running iOS 13.1.2.
+const double _kOpenScale = 1.05;
+
+typedef _DismissCallback = void Function(
+  BuildContext context,
+  double scale,
+  double opacity,
+);
+
+// Given a GlobalKey, return the Rect of the corresponding RenderBox's
+// paintBounds in global coordinates.
+Rect _getRect(GlobalKey globalKey) {
+  assert(globalKey.currentContext != null);
+  final RenderBox renderBoxContainer = globalKey.currentContext.findRenderObject();
+  final Offset containerOffset = renderBoxContainer.localToGlobal(renderBoxContainer.paintBounds.topLeft);
+  return containerOffset & renderBoxContainer.paintBounds.size;
+}
+
+// The context menu arranges itself slightly differently based on the location
+// on the screen of [ContextMenu.child] before the [ContextMenu] opens.
+enum _ContextMenuLocation {
+  center,
+  left,
+  right,
+}
+
+/// A full-screen modal route that opens when the [child] is long-pressed.
+///
+/// When open, the ContextMenu shows the child, or [preview] if given, in a
+/// large full-screen Overlay with a list of buttons specified by [actions]. The
+/// child/preview is placed in an Expanded widget so that it will grow to fill
+/// the Overlay if its size is unconstrained.
+///
+/// When closed, the ContextMenu simply displays the child as if the ContextMenu
+/// were not there. Sizing and positioning is unaffected. The menu can be closed
+/// like other [PopupRoute]s, such as by tapping the background or by calling
+/// `Navigator.pop(context)`. Unlike PopupRoute, it can also be closed by
+/// swiping downwards.
+///
+/// The [preview] parameter is most commonly used to display a slight variation
+/// of [child]. See [preview] for an example of rounding the child's corners and
+/// allowing its aspect ratio to expand, similar to the iPhoto app on iOS.
```

Comment:
nit: I think the app is just called "Photos".

Reply:
Apparently iPhoto is the Mac desktop photo editing app discontinued in 2015, whoops :) Good catch.